go/ai/generate.go

```diff
@@ -172,6 +174,18 @@ func GenerateWithRequest(ctx context.Context, r *registry.Registry, opts *Genera
 		return nil, err
 	}
 
+	metadata, err := modelMetadata(r, opts.Model)
```

Comment:
Any reason why this can't be in `validateSupport` in `model_middleware.go`?

Reply:
Hmm no particular reason, I was kind of biased by the constrained generation check we do at this place and wanted to make use of it as well. But definitely makes more sense to place it in the middleware 🤔